server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -136,16 +144,92 @@ public GlobalOrdinalsStringTermsAggregator(
                 return new DenseGlobalOrds();
             });
         }
+        this.fieldName = (valuesSource instanceof ValuesSource.Bytes.WithOrdinals.FieldData)
+            ? ((ValuesSource.Bytes.WithOrdinals.FieldData) valuesSource).getIndexFieldName()
+            : null;
     }
 
     String descriptCollectionStrategy() {
         return collectionStrategy.describe();
     }
 
+    public void setWeight(Weight weight) {
+        this.weight = weight;
+    }
+
+    /**
+     Collects term frequencies for a given field from a LeafReaderContext.
+     @param ctx The LeafReaderContext to collect terms from
+     @param globalOrds The SortedSetDocValues for the field's ordinals
+     @param ordCountConsumer A consumer to accept collected term frequencies
+     @return A no-operation LeafBucketCollector implementation, since collection is complete
+     @throws IOException If an I/O error occurs during reading
+     */
+    LeafBucketCollector termDocFreqCollector(
+        LeafReaderContext ctx,
+        SortedSetDocValues globalOrds,
+        BiConsumer<Long, Integer> ordCountConsumer
+    ) throws IOException {
+        if (weight == null || weight.count(ctx) != ctx.reader().maxDoc()) {
```

Comment:
I suppose this also checks whether there are deletions in this segment? worth a comment for that.

or better to explicitly check `weight.count(ctx) != -1`

Reply:
I did not added the condition `weight.count(ctx) != -1` explicitly because that will be implicit in `weight.count(ctx) != ctx.reader().maxDoc()` since maxDoc will always return a non-negative number.

`weight.count(ctx)` takes into account deletions when matching docs, so if that matches maxdoc() it implies no docs are deleted here, but yes, can include this in comment as well.